compiler/rustc_builtin_macros/src/env.rs

```diff
@@ -176,3 +183,50 @@ fn is_cargo_env_var(var: &str) -> bool {
         || var.starts_with("DEP_")
         || matches!(var, "OUT_DIR" | "OPT_LEVEL" | "PROFILE" | "HOST" | "TARGET")
 }
+
+const KNOWN_CARGO_VARS: &[&str] = &[
+    // List of known Cargo environment variables that are set for crates (not build scripts, OUT_DIR etc).
+    // See: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates
```

Comment:
Discussion: I have two reservations about making this suggestion:

1. Wouldn't this depend on the specific cargo version?
2. Basically, what I was trying to get at, is that _in principle_ I feel like `rustc` should not make assumptions about `cargo` (that is, `rustc` should not "know" about `cargo`'s existence), because `cargo` isn't the only way to invoke `rustc`

Reply:
Yeah, seems alright, was just curious to hear what you think.